File: Leetcode/2799. Count Complete Subarrays in an Array.py
# https://leetcode.com/problems/count-complete-subarrays-in-an-array/

from typing import List


class Solution:
    def countCompleteSubarrays(self, nums: List[int]) -> int:
        result = 0
        totalDistinct = len(set(nums))
        for start in range(len(nums)):
            distinct = set()
            for i in range(start, len(nums)):
                if nums[i] not in distinct:
                    distinct.add(nums[i])
                if len(distinct) == totalDistinct:
                    result += 1
        return result

File: Leetcode/2001-3000/2701-2800/2799. Count Complete Subarrays in an Array.go
func countCompleteSubarrays(nums []int) int {
    result := 0
    distincts := map[int]bool{}
    for _, num := range nums {
        if _, exist := distincts[num]; !exist {
            distincts[num] = true;
        }
    }
    totalDistincts := len(distincts)
    for start := 0; start < len(nums); start++ {
        distinct := map[int]bool{}
        for i := start; i < len(nums); i++ {
            if _, exist := distinct[nums[i]]; !exist {
                distinct[nums[i]] = true
            }
            if len(distinct) == totalDistincts{
                result++
            }
        }
    }
    return result
}